src/NLog.StructuredLogging.Json/LoggerExtensions.cs

```diff
@@ -91,11 +93,27 @@ private static void TransferDataToLogEventProperties(LogEventInfo log, object lo
                 return;
             }
 
-            logProperties.GetType()
-                .GetProperties()
-                .Where(p => p.GetIndexParameters().Length == 0)
-                .ToList()
-                .ForEach(x => log.Properties.Add(x.Name, x.GetValue(logProperties)));
+            if (IsDictionary(logProperties))
+            {                
+                foreach (var logProperty in (IDictionary<object, object>) logProperties)
```

Comment:
We expect the key to be a string - should we check that it is a dictionary of <`string, object>`, or just do a `.ToString` on the key object?